spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/CompositeHealthIndicator.java

```diff
@@ -63,11 +64,39 @@ public void addHealthIndicator(String name, HealthIndicator indicator) {
 
 	@Override
 	public Health health() {
+		StopWatch stopWatch = new StopWatch();
+		stopWatch.start();
+
 		Map<String, Health> healths = new LinkedHashMap<>();
 		for (Map.Entry<String, HealthIndicator> entry : this.indicators.entrySet()) {
-			healths.put(entry.getKey(), entry.getValue().health());
+			healths.put(entry.getKey(), healthIndicatorWithResponseTime(entry.getValue()));
 		}
-		return this.healthAggregator.aggregate(healths);
+
+		Health aggregateHealths = this.healthAggregator.aggregate(healths);
+
+		stopWatch.stop();
+
+		return addResponseTimeMsToHealth(aggregateHealths, stopWatch);
+	}
+
+	private Health healthIndicatorWithResponseTime(HealthIndicator healthIndicator) {
+		StopWatch stopWatch = new StopWatch();
```

Comment:
You're right with the short lived objects, I've updated to make it a bit clearer for utilizing StopWatch and reducing the amount of objects created